storage/remote/googleiam/googleiam.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googleiam
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+	apihttp "google.golang.org/api/transport/http"
+)
+
+type GoogleIAMConfig struct {
```

Comment:
Linter is right we could do `Config` here (and `NewRT` below or something)

Reply:
I tried to keep it consistent with aws and azure. They use the same format https://github.com/prometheus/prometheus/blob/main/storage/remote/azuread/azuread.go#L71-L83. Let me know if you want me to change it.

Reply:
I would change it. Just `Config` and `NewRT`